### PR TITLE
UCP/FLUSH: Remove extra progress deregistration

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Remove extra progress deregistration

## Why ?

`ucp_ep_flush_slow_path_remove()` and `ucp_flush_check_completion()` do registration of `ucp_ep_flush_resume_slow_path_callback()` CB

## How ?

1. Remove `ucp_ep_flush_slow_path_remove()` wrapper
2. Do not deregister the CB in `ucp_ep_flush_resume_slow_path_callback()` - this will be done in `ucp_flush_check_completion()`

### CODE REVIEW

**Comment by dmitrygx** [comment]
#5210
bot:pipe:retest

File: src/ucp/rma/flush.c [hunk 1]
```diff
@@ -167,7 +161,6 @@ static unsigned ucp_ep_flush_resume_slow_path_callback(void *arg)
 {
     ucp_request_t *req = arg;
 
-    ucp_ep_flush_slow_path_remove(req);
```
Comment (yosefe) [inline src/ucp/rma/flush.c:170]: what if ucp_flush_check_completion() returns 0, seems like progress will not be removed?

Comment (dmitrygx) [reply src/ucp/rma/flush.c:170]: yes, you're right
should we use `UCS_CALLBACKQ_FLAG_ONESHOT` flag?

Comment (yosefe) [reply src/ucp/rma/flush.c:170]: maybe, pls check that it will keep current behavior

Comment (dmitrygx) [reply src/ucp/rma/flush.c:170]: ok

**Pull request closed without merging**